packages/flutter/lib/src/rendering/object.dart

```diff
@@ -4585,652 +4353,1109 @@ mixin RelayoutWhenSystemFontsChangeMixin on RenderObject {
+    // node, thus marking this semantics boundary dirty is not enough, it needs
+    // to find the first parent semantics boundary that does not have any
+    // possible sibling node.
+    while (node.semanticsParent != null && (mayProduceSiblingNodes || !isEffectiveSemanticsBoundary)) {
+      if (node != renderObject && node._semantics.parentDataDirty && !mayProduceSiblingNodes) {
+        break;
+      }
+      node._semantics.parentData = null;
+      node._semantics._blocksPreviousSibling = null;
+      node._semantics.elevationAdjustment = 0.0;
+      // Since this node is a semantics boundary, the produced sibling nodes will
+      // be attached to the parent semantics boundary. Thus, these sibling nodes
+      // will not be carried to the next loop.
+      if (isEffectiveSemanticsBoundary) {
+        mayProduceSiblingNodes = false;
+      }
+      mayProduceSiblingNodes |= node._semantics.configProvider.effective.childConfigurationsDelegate != null;
+
+      node = node.semanticsParent!;
+      // If node._semantics.built is false, this branch is currently blocked.
+      // In that case, it should continue dirty upward until it reach a
+      // unblocked semantics boundary because blocked branch will not rebuild
+      // semantics during PipelineOwner.flushSemantics.
+      //
+      // If we stop here and not dirty the rendering parent and a flush semantics
+      // is called, it will end up with a blocked branch where some sub-branch
+      // is dirty. There won't be a way to rebuild these dirty sub-branch
+      // without rebuilding the entire blocked branch (which is costly) when the
+      // branch is later unblocked.
+      isEffectiveSemanticsBoundary =
+          node._semantics.configProvider.effective.isSemanticBoundary
+          && node._semantics.built;
+    }
+    if (node != renderObject && producedSemanticsNode != null && node._semantics.parentDataDirty) {
+      // If `this` node has already been added to [owner._nodesNeedingSemantics]
+      // remove it as it is no longer guaranteed that its semantics
+      // node will continue to be in the tree. If it still is in the tree, the
+      // ancestor `node` added to [owner._nodesNeedingSemantics] at the end of
+      // this block will ensure that the semantics of `this` node actually gets
+      // updated.
+      // (See semantics_10_test.dart for an example why this is required).
+      renderObject.owner!._nodesNeedingSemantics.remove(renderObject);
+    }
+    if (!node._semantics.parentDataDirty) {
+      if (renderObject.owner != null) {
+        assert(node._semantics.configProvider.effective.isSemanticBoundary || node.semanticsParent == null);
+        if (renderObject.owner!._nodesNeedingSemantics.add(node)) {
+          renderObject.owner!.requestVisualUpdate();
+        }
+      }
+    }
+  }
+
+  void _marksExplicitInMergeGroup(List<_SemanticsFragment> mergeGroup, {bool isMergeUp = false}) {
```

Comment:
nit: all of the code below is dealing with "conflicts", and at the end it calls `markSiblingConfigurationConflict` for the conflicting fragments. Should we call this method `_markConflictsInMergeGroup(...)`?